BlackDTestCase.test_cors_preflight failure on 22.8.0

<!--
Please make sure that the bug is not already fixed either in newer versions or the
current development version. To confirm this, you have three options:

1. Update Black's version if a newer release exists: `pip install -U black`
2. Use the online formatter at <https://black.vercel.app/?version=main>, which will use
   the latest main branch.
3. Or run _Black_ on your machine:
   - create a new virtualenv (make sure it's the same Python version);
   - clone this repository;
   - run `pip install -e .[d]`;
   - run `pip install -r test_requirements.txt`
   - make sure it's sane by running `python -m pytest`; and
   - run `black` like you did last time.
-->

**Describe the bug**

BlackDTestCase.test_cors_preflight test failure:

<details>
  <summary>show/hide</summary>

```
______________________ BlackDTestCase.test_cors_preflight ______________________

self = <tests.test_blackd.BlackDTestCase testMethod=test_cors_preflight>

    async def get_application(self) -> web.Application:
>       return blackd.make_app()

tests/test_blackd.py:46:
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
test-env/lib/python3.10/site-packages/blackd/__init__.py:76: in make_app
    executor = ProcessPoolExecutor()
/usr/lib/python3.10/concurrent/futures/process.py:657: in __init__
    self._call_queue = _SafeQueue(
/usr/lib/python3.10/concurrent/futures/process.py:168: in __init__
    super().__init__(max_size, ctx=ctx)
/usr/lib/python3.10/multiprocessing/queues.py:42: in __init__
    self._reader, self._writer = connection.Pipe(duplex=False)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

duplex = False

    def Pipe(duplex=True):
        '''
        Returns pair of connection objects at either end of a pipe
        '''
        if duplex:
            s1, s2 = socket.socketpair()
            s1.setblocking(True)
            s2.setblocking(True)
            c1 = Connection(s1.detach())
            c2 = Connection(s2.detach())
        else:
>           fd1, fd2 = os.pipe()
E           OSError: [Errno 24] Too many open files

/usr/lib/python3.10/multiprocessing/connection.py:532: OSError
=========================== short test summary info ============================
FAILED tests/test_blackd.py::BlackDTestCase::test_cors_preflight - OSError: [...
================== 1 failed, 286 passed, 3 skipped in 11.32s ===================
```
</details>

**To Reproduce**

<!--
Minimal steps to reproduce the behavior with source code and Black's configuration.
-->

https://github.com/archlinux/svntogit-community/blob/8ef0305f6c9b28df533696a7e54d08269dc75aa8/trunk/PKGBUILD#L33-L35

```
python -m venv --system-site-packages test-env
test-env/bin/python -m installer dist/*.whl
PATH="$PWD/test-env/bin:$PATH" test-env/bin/python -m pytest
```

The package build process also hangs after the above output.

**Expected behavior**

<!-- A clear and concise description of what you expected to happen. -->

Test passes and we all live happily ever after.

**Environment**

<!-- Please complete the following information: -->

- Black's version: 22.8.0
- OS and Python version: <!-- e.g. [Linux/Python 3.7.4rc1] --> Arch Linux/Python 3.10.6

**Additional context**

<!-- Add any other context about the problem here. -->

Can provide dependency versions if that would help.

--- a/src/blackd/__init__.py
+++ b/src/blackd/__init__.py
@@ -2,7 +2,7 @@ import asyncio
 import logging
 from concurrent.futures import Executor, ProcessPoolExecutor
 from datetime import datetime, timezone
-from functools import partial, cache
+from functools import cache, partial
 from multiprocessing import freeze_support
 
 try:

--- a/src/blackd/__init__.py
+++ b/src/blackd/__init__.py
@@ -85,7 +85,6 @@ def main(bind_host: str, bind_port: int) -> None:
     web.run_app(app, host=bind_host, port=bind_port, handle_signals=True, print=None)
 
 
-
 @cache
 def executor() -> Executor:
     return ProcessPoolExecutor()

--- a/src/blackd/__init__.py
+++ b/src/blackd/__init__.py
@@ -2,7 +2,7 @@ import asyncio
 import logging
 from concurrent.futures import Executor, ProcessPoolExecutor
 from datetime import datetime, timezone
-from functools import partial
+from functools import partial, cache
 from multiprocessing import freeze_support
 
 try:
@@ -85,12 +85,17 @@ def main(bind_host: str, bind_port: int) -> None:
     web.run_app(app, host=bind_host, port=bind_port, handle_signals=True, print=None)
 
 
+
+@cache
+def executor() -> Executor:
+    return ProcessPoolExecutor()
+
+
 def make_app() -> web.Application:
     app = web.Application(
         middlewares=[cors(allow_headers=(*BLACK_HEADERS, "Content-Type"))]
     )
-    executor = ProcessPoolExecutor()
-    app.add_routes([web.post("/", partial(handle, executor=executor))])
+    app.add_routes([web.post("/", partial(handle, executor=executor()))])
     return app
 
 

